### PR TITLE
feat: Migrate query docs to v2 layout

### PR DESCRIPTION
**Pull request by lachlancollins**
Uses new v2 docs layout and logic

See also: https://github.com/TanStack/query/pull/6763 (main), https://github.com/TanStack/query/pull/6764 (v4), https://github.com/TanStack/query/pull/6765 (v3)

### CODE REVIEW

File: app/projects/query.tsx
Comment (fulopkovacs) [inline app/projects/query.tsx:35]: Just to be consistent with code 3 lines above:
```suggestion
    <Link to=".." className="font-bold">
```

**Review by fulopkovacs — CHANGES_REQUESTED** [review]
One tiny comment, but otherwise it's good to go! No need to request a second review (unless GitHub requires it). ☺️

File: app/projects/query.tsx
Comment (lachlancollins) [reply app/projects/query.tsx]: Fixed - it does need another review, sorry!

I'll do this for the other PRs too

**Review by fulopkovacs — APPROVED** [review]
let's goo! 🚢

**Pull request merged**
